pkg/common/jobcontroller/jobcontroller.go

```diff
@@ -197,6 +199,34 @@ func (jc *JobController) GenLabels(jobName string) map[string]string {
 	}
 }
 
+func (jc *JobController) SyncPodGroup(job metav1.Object, minAvailableReplicas int32) (*v1alpha1.PodGroup, error) {
+
+	kubeBatchClientInterface := kbutil.GetKubeBatchClientInterface()
+	// Check whether podGroup exists or not
+	podGroup, err := kubeBatchClientInterface.SchedulingV1alpha1().PodGroups(job.GetNamespace()).Get(job.GetName(), metav1.GetOptions{})
+	if err == nil || !k8serrors.IsNotFound(err) {
+		if err == nil {
+			err = errors.New(string(metav1.StatusReasonAlreadyExists))
+		}
+		return podGroup, err
```

Comment:
Do we need to return error if it already exists? Isn't this will be called for every reconcile ?

Reply:
In SyncPdb method it has been handled like this, so I have tried to mimic what SyncPdb was doing, but instead of PDB, have handled PodGroup

Reply:
I think, we missed this earlier. I don't see any need to return error. @richardsliu

Reply:
Okay. Will change that.